src/osdatahub/NamesAPI/names_api.py

```diff
@@ -0,0 +1,160 @@
+            limit (int, optional): The maximum number of features to return.
+                Defaults to 100.
+            bounds (Extent, optional): Biases the results to a certain area. Must be British National Grid
+                (EPSG:27700) CRS
+            bbox_filter (Extent, optional): Filters the results to a certain area. Must be British National Grid
+                (EPSG:27700) CRS
+            local_type (Union[Iterable, str], optional): Filters the results to certain local types. Available local
+                types can be found at the bottom of https://osdatahub.os.uk/docs/names/technicalSpecification
+
+        Returns:
+            FeatureCollection: The results of the query in GeoJSON format
+        """
+        data = GrowList()
+        params = {"query": text}
+
+        if limit <= 0:
+            raise ValueError("Parameter \"limit\" must be a positive integer")
+
+        if bounds:
+            if not bounds.crs == "EPSG:27700":
+                raise TypeError("Bounds must be in British National Grid CRS (EPSG:27700)")
+            params.update({"bounds": bounds.bbox.to_string()})
+        if bbox_filter or local_type:
+            if bbox_filter and bbox_filter.crs != "EPSG:27700":
+                raise TypeError("Bounding Box filter must be in British National Grid CRS (EPSG:27700)")
+            params.update({"fq": self.__format_fq(bbox_filter, local_type)})
+
+        try:
+            n_required = min(limit, 100)
+            while n_required > 0 and data.grown:
+                params.update({"offset": len(data), "maxresults": n_required})
+                print(params)
+                response = requests.get(self.__endpoint("find"), params=params)
+                data.extend(self.__format_response(response))
+                n_required = min(100, limit - len(data))
+        except KeyError:
+            response.raise_for_status()
+        return addresses_to_geojson(data.values, "EPSG:27700")
+
+    def nearest(self, point: tuple, radius: float = 100, local_type: Union[Iterable, str] = None) -> FeatureCollection:
+        """Takes a pair of coordinates (X, Y)/(Lon, Lat) as an input
+        to determine the closest name.
+
+        Args:
+            point (tuple): A set of coordinates in British National Grid (EPSG:27700) format
+            radius (float): The search radius in metres (max. 1000).
+                Defaults to 100.
+            local_type (Union[Iterable, str], optional):  Filters the results to certain local types. Available local
+                types can be found at the bottom of https://osdatahub.os.uk/docs/names/technicalSpecification
+
+        Returns:
+            FeatureCollection: The results of the query in GeoJSON format
+        """
+        data = GrowList()
+        if not all([str(p).isnumeric() for p in point]):
+            raise TypeError("All values in argument \"point\" must be numeric")
+        if not 0.01 <= radius <= 1000:
```

Comment:
Can radius equal 0.001, in which case this error message might confuzzle the  user

Reply:
radius cannot equal less than 0.01. Changed the error message to resolve